### PR TITLE
[mlir][tosa] Add folding for TOSA ArgMax operator

### PR DESCRIPTION
**Pull request by d-agbv**
TOSA ArgMax operator could be folded into a constant tensor filled with zeros when dimension of the selected axis equals one.

### CODE REVIEW

File: mlir/lib/Dialect/Tosa/IR/TosaCanonicalizations.cpp [hunk 1]
```diff
@@ -507,6 +507,20 @@ OpFoldResult AddOp::fold(FoldAdaptor adaptor) {
...
+OpFoldResult ArgMaxOp::fold(FoldAdaptor adaptor) {
+  auto inputTy = llvm::dyn_cast<RankedTensorType>(getInput().getType());
+  auto outputTy = llvm::dyn_cast<RankedTensorType>(getType());
+  if (!inputTy || !outputTy || !inputTy.hasStaticShape() ||
+      !outputTy.hasStaticShape())
+    return {};
+
+  if (inputTy.getDimSize(getAxis()) == 1) {
+    return DenseElementsAttr::get(outputTy, 0);
+  }
```
Comment (kuhar) [inline mlir/lib/Dialect/Tosa/IR/TosaCanonicalizations.cpp:519]: nit: for consistency with the if above, I'd either drop these braces or also add them above

**Review by kuhar — APPROVED** [review]

File: mlir/test/Dialect/Tosa/constant-op-fold.mlir [hunk 2]
```diff
@@ -3,6 +3,20 @@
 
 // RUN: mlir-opt --split-input-file --tosa-layerwise-constant-fold="aggressive-reduce-constant=true" %s | FileCheck %s --check-prefix=AGGRESIVE
 
+// CHECK-LABEL: @argmax_fold_dim_1
```
Comment (eric-k256) [inline mlir/test/Dialect/Tosa/constant-op-fold.mlir:6]: nit: I thought from the name this was looking at dimension 1, not a dimension of size 1. maybe @armax_fold_dim_size_1 instead?

**Review by eric-k256 — COMMENTED** [review]
The code itself looks fine, one minor request, especially if you're also fixing the brackets to make the test name a bit easier to understand.
Thanks for the contribution.

File: mlir/lib/Dialect/Tosa/IR/TosaCanonicalizations.cpp (hunk 1 above)
Comment (d-agbv) [reply mlir/lib/Dialect/Tosa/IR/TosaCanonicalizations.cpp]: Thank you for the review! I updated the code and dropped braces here.

**Review by GeorgeARM — APPROVED** [review]

**Review by eric-k256 — APPROVED** [review]

**Pull request merged**